relay-event-normalization/src/normalize/span/description/mod.rs

```diff
@@ -193,51 +195,106 @@ fn scrub_file(description: &str) -> Option<String> {
     }
 }
 
-/// Normalize the domain and port of a URL.
+/// Scrub a `Url::Host` object. For domain names, run the string through a scrubber. For IP addresses, return the IP address as-is.
+///
+/// # Arguments
+///
+/// * `host` - `Url::Host`
+///
+/// # Returns
+///
+/// A host `String`, or `None` if scrubbing fails.
+pub fn scrub_host(host: Host<&str>) -> String {
+    match host {
+        Host::Ipv4(ip) => scrub_ipv4(ip),
+        Host::Ipv6(ip) => scrub_ipv6(ip),
+        Host::Domain(domain) => scrub_domain_name(String::from(domain)),
+    }
+}
+
+/// Scrub an IPV4 address. Allow well-known IPs like loopback, and fully scrub out all other IPs.
+///
+/// # Arguments
+///
+/// * `ip` - IP to be scrubbed
+///
+/// # Returns
+///
+/// Scrubbed string
+pub fn scrub_ipv4(ip: Ipv4Addr) -> String {
+    if IPV4_ALLOW_LIST.contains(&ip.to_string().as_str()) {
+        return ip.to_string();
+    }
+
+    String::from("*.*.*.*")
+}
+
+const IPV4_ALLOW_LIST: [&str; 1] = ["127.0.0.1"];
```

Comment:
Let's make this an array of `Ipv4Addr`, so the check does not have to allocate a `String` (same for IPv6).